### PR TITLE
[material-ios] Updated the binding to v72.2.0

### PR DESCRIPTION
**Pull request by mattleibow**
Updated to https://github.com/material-components/material-components-ios/releases/tag/v72.1.0

> VS bug [#747237](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/747237)

### CODE REVIEW

**Comment by PureWeen** [comment]
Version 72 Installed and ran fine with Visual Gallery locally
Only thing I had to change was the call to ```SetShadowElevation``` which now takes an nfloat opposed to a double

**Comment by mattleibow** [comment]
Yeah, they changed the type of all things elevated to `CGFloat` or our `nfloat`.

File: iOS/MaterialComponents/source/MaterialComponents/ApiDefinition.cs [hunk 1]
```diff
@@ -453,33 +661,126 @@ interface AppBar {
 		bool InferTopSafeAreaInsetFromViewController { get; set; }
 	}
 
-	// @interface MDCAppBarTextColorAccessibilityMutator : NSObject
-	[Obsolete ("This API will be deprecated with no replacement.")]
-	[BaseType (typeof (NSObject), Name = "MDCAppBarTextColorAccessibilityMutator")]
-	interface AppBarTextColorAccessibilityMutator {
```
Comment (mattleibow) [inline iOS/MaterialComponents/source/MaterialComponents/ApiDefinition.cs:459]: The mutator is dead!

File: iOS/MaterialComponents/source/MaterialComponents/ApiDefinition.cs [hunk 2]
```diff
@@ -453,33 +661,126 @@ interface AppBar {
...
+		void ApplyTypographyScheme (ITypographyScheming typographyScheme, SelfSizingStereoCell cell);
 	}
 
 	// @interface MDCBaseCell : UICollectionViewCell
 	[BaseType (typeof (UICollectionViewCell), Name = "MDCBaseCell")]
 	interface BaseCell {
 		// @property (assign, nonatomic) MDCShadowElevation elevation;
 		[Export ("elevation")]
-		double Elevation { get; set; }
+		nfloat Elevation { get; set; }
```
Comment (mattleibow) [inline iOS/MaterialComponents/source/MaterialComponents/ApiDefinition.cs:738]: The `MDCShadowElevation` has been mutated into a `CGFloat`

File: iOS/MaterialComponents/source/MaterialComponents/ApiDefinition.cs [hunk 3]
```diff
@@ -945,8 +1434,15 @@ interface Button : INativeObject {
 	}
 
 	// @interface MDCButtonBar : UIView
-	[BaseType (typeof (UIView), Name = "MDCButtonBar")]
+	[BaseType (typeof (UIView),
+		Name = "MDCButtonBar",
+		Delegates = new [] { "Delegate" },
+		Events = new [] { typeof (ButtonBarDelegate) })]
```
Comment (mattleibow) [inline iOS/MaterialComponents/source/MaterialComponents/ApiDefinition.cs:1452]: They added a `delegate` property, so I added our magic.

File: iOS/MaterialComponents/source/MaterialComponents/ApiDefinition.cs [hunk 4]
```diff
@@ -4777,110 +5411,105 @@ interface RoundedCornerTreatment {
...
 		UIColor OnBackgroundColor { get; }
 	}
 
-	// @interface MDCSemanticColorScheme : NSObject <MDCColorScheming>
+	// @interface MDCSemanticColorScheme : NSObject <MDCColorScheming, NSCopying>
 	[BaseType (typeof (NSObject), Name = "MDCSemanticColorScheme")]
-	interface SemanticColorScheme : ColorScheming {
-		// @property (readwrite, nonatomic) UIColor * _Nonnull primaryColor;
-		[Export ("primaryColor", ArgumentSemantic.Assign)]
+	interface SemanticColorScheme : ColorScheming, INSCopying {
```
Comment (mattleibow) [inline iOS/MaterialComponents/source/MaterialComponents/ApiDefinition.cs:5486]: Some schemes are now copy-stuff.

File: iOS/MaterialComponents/source/MaterialComponents/ApiDefinition.cs [hunk 5]
```diff
@@ -4893,101 +5522,6 @@ interface SemanticColorScheme : ColorScheming {
 		UIColor BlendColor (UIColor color, UIColor backgroundColor);
 	}
 
-	[Static]
-	interface ShadowElevations {
```
Comment (mattleibow) [inline iOS/MaterialComponents/source/MaterialComponents/ApiDefinition.cs:4897]: This whole type has been moved to a const enum, and no longer fields

File: iOS/MaterialComponents/source/MaterialComponents/ApiDefinition.cs [hunk 6]
```diff
@@ -5080,22 +5618,13 @@ interface ShapedView {
 		[Export ("initWithFrame:shapeGenerator:")]
 		[DesignatedInitializer]
 		IntPtr Constructor (CGRect frame, [NullAllowed] IShapeGenerating shapeGenerator);
-
-		// -(instancetype _Nonnull)initWithFrame:(CGRect)frame;
-		[Export ("initWithFrame:")]
-		IntPtr Constructor (CGRect frame);
```
Comment (mattleibow) [inline iOS/MaterialComponents/source/MaterialComponents/ApiDefinition.cs:5086]: This was removed, and I added it back in the extensions file

File: iOS/MaterialComponents/source/MaterialComponents/Extensions.cs [hunk 7]
```diff
@@ -2,11 +2,89 @@
...
 using System.Runtime.InteropServices;
 
+using CoreGraphics;
 using Foundation;
 using ObjCRuntime;
 using UIKit;
 
 namespace MaterialComponents {
+
+	public static class ShadowElevations {
```
Comment (mattleibow) [inline iOS/MaterialComponents/source/MaterialComponents/Extensions.cs:12]: The new consts

File: iOS/MaterialComponents/source/MaterialComponents/Extensions.cs [hunk 8]
```diff
@@ -35,20 +113,22 @@ public static class MaterialFeatureHighlightStrings {
...
 
 	public partial class CardScheme {
-		SemanticColorScheme semanticColorScheme;
 		public SemanticColorScheme SemanticColorScheme {
 			get {
-				if (semanticColorScheme == null)
-					semanticColorScheme = Runtime.GetNSObject<SemanticColorScheme> (ColorScheme.Handle, false);
-
-				return semanticColorScheme;
+				return Runtime.GetNSObject<SemanticColorScheme> (ColorScheme.Handle, false);
```
Comment (mattleibow) [inline iOS/MaterialComponents/source/MaterialComponents/Extensions.cs:183]: I may be wrong, but we shouldn't cache this as changing the other property will break this? This is supposed to be the same object - just a more type-specific version. ie: SemanticColorScheme === ColorScheme

File: iOS/MaterialComponents/source/MaterialComponents/Extensions.cs [hunk 9]
```diff
@@ -35,20 +113,22 @@ public static class MaterialFeatureHighlightStrings {
...
 			}
 			set {
-				semanticColorScheme?.Dispose ();
-				semanticColorScheme = null;
 				ColorScheme = value;
 			}
 		}
+		//public ShapeScheme ShapeScheme {
```
Comment (mattleibow) [inline iOS/MaterialComponents/source/MaterialComponents/Extensions.cs:189]: I couldn't think of another name for this - the types have changed, but are just different with a new `I`

File: iOS/MaterialComponents/source/MaterialComponents/Properties/AssemblyInfo.cs [hunk 10]
```diff
@@ -17,15 +17,16 @@
 [assembly: AssemblyConfiguration ("")]
 [assembly: AssemblyCompany ("")]
 [assembly: AssemblyProduct ("")]
-[assembly: AssemblyCopyright ("(c) Bill Holmes")]
+[assembly: AssemblyCopyright ("© Microsoft Corporation. All rights reserved.")]
```
Comment (mattleibow) [inline iOS/MaterialComponents/source/MaterialComponents/Properties/AssemblyInfo.cs:20]: @bholmes 🥇

**Comment by mattleibow** [comment]
LOL, another update :)

**Review by newky2k — APPROVED** [review]

**Pull request merged**